### PR TITLE
feat: add Composite provider

### PR DESCRIPTION
**Pull request by beeme1mr**
## Overview

The federated provider is based on the "super provider" concept that has been brought up in a number of community meetings. This PR adds an provider that supports an arbitrary number of child providers. Flags are evaluated based on the order providers are registered.

## Limitations

- All child provider hooks are run.

### CODE REVIEW

File: libs/providers/federated/src/lib/federated-provider.ts [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+    defaultValue: number,
+    context: EvaluationContext
+  ): Promise<ResolutionDetails<number>> {
+    for (const provider of this.providers) {
+      const result = await provider.resolveNumberEvaluation(
+        flagKey,
+        defaultValue,
+        context
+      );
+      if (!result.errorCode) {
```
Comment (justinabrahms) [inline libs/providers/federated/src/lib/federated-provider.ts:66]: You may want to do some logging in the error cases here.

**Review by justinabrahms — APPROVED** [review]

**Comment by justinabrahms** [comment]
optional nitpick: "Composite provider"? federated makes me think it's decentralized.

File: libs/providers/federated/src/lib/federated-provider.ts (hunk 1 above)
Comment (beeme1mr) [reply libs/providers/federated/src/lib/federated-provider.ts:66]: That's a good idea. I'll do that once [this PR](https://github.com/open-feature/node-sdk/pull/219) is merged.

**Comment by beeme1mr** [comment]
@justinabrahms, the use of federation in this context was inspired by [GraphQL](https://www.apollographql.com/docs/federation/). Happy to change it to something else if it's confusing though.

FYI @toddbaert

**Pull request closed without merging**